src/viz.py

```diff
@@ -0,0 +1,143 @@
+from utils import *
+from train import *
+from network import *
+from sklearn.preprocessing import StandardScaler
+from sklearn.decomposition import PCA
+import numpy
+
+def process_batch(batch):
+    # Pos_text is unused so don't even bother loading it up
+    text, mel, text_len, mel_len = batch
+
+    # Send stuff we use alot to device this is character, mel, mel_input, and pos_mel
+    text, mel, = text.to(DEVICE), mel.to(DEVICE)
+    text_len, mel_len = text_len.to(DEVICE), mel_len.to(DEVICE)
+
+    return (text, mel, text_len, mel_len), (None)
+
+def twod_viz(args, model=None, num_points=300, to_keep=5):
+
+    test_dataset = get_dataset('test.csv')
+    test_dataloader = DataLoader(test_dataset,
+            batch_size=num_points,
+            collate_fn=collate_fn_transformer,
+            num_workers=1)
+    _, _, model, _, _ = initialize_model(args)
+    model.eval()
+
+    for batch in test_dataloader:
+        batch = process_batch(batch)
+        x, _ = batch
+        text, mel, text_len, mel_len = x
+
+        # Get the last latent state of sequence for both text and speech
+        with torch.no_grad():
+            text_latent, _ = model.text_m.encode(text, text_len)
+            speech_latent, _ = model.speech_m.encode(mel, mel_len)
+
+        if args.model_type == 'rnn':
+            _, t_out = text_latent
+            _, s_out = speech_latent
+        else:
+            t_out = text_latent
+            s_out = speech_latent        
+        text_latent = t_out[:, -1, :]
```

Comment:
try the torch.mean(t_out, dim=1)

Reply:
oh also since these are from the same batch, the technically have the same size, could just run t-sne/pca on 
t_out directly.